site-src/api-types/grpcroute.md

```diff
@@ -0,0 +1,269 @@
+routing based on rules and filters (optional). A hostname is the fully qualified
+domain name of a network host, as defined by [RFC 3986][rfc-3986]. Note the
+following deviations from the “host” part of the URI as defined in the RFC:
+
+- IPs are not allowed.
+- The : delimiter is not respected because ports are not allowed.
+
+Incoming requests are matched against hostnames before the GRPCRoute rules are
+evaluated. If no hostname is specified, traffic is routed based on GRPCRoute
+rules and filters (optional).
+
+The following example defines hostname "my.example.com":
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: GRPCRoute
+metadata:
+  name: grpcroute-example
+spec:
+  hostnames:
+  - my.example.com
+```
+
+### Rules
+
+Rules define semantics for matching an gRPC requests based on conditions,
+optionally executing additional processing steps, and optionally forwarding
+the request to an API object.
+
+#### Matches
+
+Matches define conditions used for matching an gRPC requests. Each match is
+independent, i.e. this rule will be matched if any single match is satisfied.
+
+Take the following matches configuration as an example:
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: GRPCRoute
+...
+matches:
+  - method:
+      service: com.example.User
+      method: Login
+    headers:
+      values:
+        version: "2"
+  - method:
+      service: com.example.v2.User
+      method: Login
+```
+
+For a request to match against this rule, it must satisfy EITHER of the
+following conditions:
+
+ - The `com.example.User.Login` method **AND** contains the header "version: 2"
+ - The `com.example.v2.User.Login` method.
+
+If no matches are specified, the default is to match every gRPC request.
+
+#### Filters (optional)
+
+Filters define processing steps that must be completed during the request or
+response lifecycle. Filters act as an extension point to express additional
+processing that may be performed in Gateway implementations. Some examples
+include request or response modification, implementing authentication
+strategies, rate-limiting, and traffic shaping.
+
+The following example adds header "my-header: foo" to gRPC requests with Host
+header "my.filter.com". Note that GRPCRoute uses HTTPRoute filters for features
+with functionality identical to HTTPRoute, such as this.
+
+```yaml
+{% include 'experimental/grpc-filter.yaml' %}
+```
+
+API conformance is defined based on the filter type. The effects of ordering
+multiple behaviors are currently unspecified. This may change in the future
+based on feedback during the alpha stage.
+
+Conformance levels are defined by the filter type:
+
+ - All "core" filters MUST be supported by implementations supporting GRPCRoute.
+ - Implementers are encouraged to support "extended" filters.
+ - "Custom" filters have no API guarantees across implementations.
+
+Specifying a core filter multiple times has unspecified or custom conformance.
```

Comment:
Tiny nit: After https://github.com/kubernetes-sigs/gateway-api/pull/1436 merged, "custom" should be replaced by "Implementation-Specific"

Reply:
:+1: